File: heaps/minimum_cost_to_connect_sticks.java
public int connectSticks(int[] sticks)
{
    int totalCost = 0;

    PriorityQueue<Integer> pq = new PriorityQueue<>();

    // Add all sticks to min heap
    for (int stick : sticks)
    {
        pq.add(stick);
    }

    // Combine two of smallest sticks until left with just one
    while (pq.size() > 1)
    {
        int stick1 = pq.remove();
        int stick2 = pq.remove();

        int cost = stick1 + stick2;
        totalCost += cost;

        pq.add(cost);
    }

    return totalCost;
}

File: heaps/minimum_cost_to_connect_sticks.py
from heapq import *


def connectSticks(sticks: list[int]) -> int:
    total_cost = 0

    min_heap = sticks
    heapify(min_heap)

    while len(min_heap) > 1:
        stick1, stick2 = heappop(min_heap), heappop(min_heap)

        cost = stick1 + stick2
        total_cost += cost

        heappush(min_heap, cost)

    return total_cost